backend/src/services/trading_service.py

```
@@ -175,57 +204,153 @@ async def execute_sell_order(self, token_address: str, amount_tokens: float, sli
+                        logger.error(f"Transaction failed: {status.err}")
+                        return False
+                    if status.confirmation_status in ['confirmed', 'finalized']:
+                        return True
+            except Exception as e:
+                logger.debug(f"Error polling signature status: {e}")
+
+            await asyncio.sleep(retry_interval)
+            retry_interval = min(retry_interval * 1.2, 10.0) # Adaptive polling
+        return False
+
+    async def place_limit_order(self, token_address: str, target_price: float, amount_sol: float, side: str = 'buy', token_symbol: str = None) -> Dict:
```

Ruff (0.15.12)

[warning] PEP 484 prohibits implicit `Optional`

Convert to `T | None`

(RUF013)